### PR TITLE
feat!: add support for expiration days in storage module

### PR DESCRIPTION
**Pull request by umairidris**

### CODE REVIEW

File: examples/storage/project/versions.tf [hunk 1]
```diff
@@ -20,5 +20,5 @@ terraform {
 }
 
 provider "google" {
-  version = "~> 3.36.0"
+  version = "~> 3.0"
```
Comment (bharathkkb) [inline examples/storage/project/versions.tf:23]: We usually keep these pinned to minor versions to prevent tests from breaking too often.

**Review by bharathkkb — APPROVED** [review]
LGTM

File: examples/storage/project/versions.tf (hunk 1 above)
Comment (umairidris) [reply examples/storage/project/versions.tf]: Sounds good. I updated the version to .41 because in .38 the uniform bucket policy field was introduced.

Comment (bharathkkb) [reply examples/storage/project/versions.tf]: @umairidris could we also update [required_providers](https://github.com/terraform-google-modules/terraform-google-log-export/blob/master/modules/storage/versions.tf) accordingly?

Comment (umairidris) [reply examples/storage/project/versions.tf]: Good catch, done. Do I need to make this PR a breaking release as well?

Comment (bharathkkb) [reply examples/storage/project/versions.tf]: We had some troubles with https://github.com/terraform-google-modules/terraform-google-project-factory/issues/462 so I would say its safer to do a breaking release. We can bundle it together with https://github.com/terraform-google-modules/terraform-google-log-export/pull/64 ?

Comment (umairidris) [reply examples/storage/project/versions.tf]: Sounds good.

**Pull request merged**